Core/GDCore/IDE/Events/EventsPropertyReplacer.cpp

```diff
@@ -118,17 +116,24 @@ class GD_CORE_API ExpressionPropertyReplacer
   }
   void OnVisitVariableBracketAccessorNode(
       VariableBracketAccessorNode& node) override {
+    bool isGrandParentTypeAVariable = isParentTypeAVariable;
+    isParentTypeAVariable = false;
     node.expression->Visit(*this);
+    isParentTypeAVariable = isGrandParentTypeAVariable;
     if (node.child) node.child->Visit(*this);
   }
   void OnVisitIdentifierNode(IdentifierNode& node) override {
+    if (isParentTypeAVariable) {
+        // Do nothing, it's a variable.
+        return;
+    }
+
     auto& propertiesContainersList =
         projectScopedContainers.GetPropertiesContainersList();
 
-    // Match the potential *new* name of the property, because refactorings are
-    // done after changes in the variables container.
     projectScopedContainers.MatchIdentifierWithName<void>(
```

Comment:
I don't understand why this "MatchIdentifierWithName" is not doing its job? If it's a variable, it should match it (and so do nothing), no?

Reply:
I don't understand what you mean.
What make you think it doesn't match variables?

Reply:
oh ok, I understand. I added this before removing the "unknown" case.

Reply:
I think `isParentTypeAVariable` is still required because when a node is an identifier, we don't want to replace it if the expression is a variable parameter even if the variable doesn't exist.